runtime/dotnet/azurewebapp/Startup.cs

```diff
@@ -123,8 +129,11 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton<ICredentialProvider, ConfigurationCredentialProvider>();
             services.AddSingleton<BotAdapter>(sp => (BotFrameworkHttpAdapter)sp.GetService<IBotFrameworkHttpAdapter>());
 
-            // Register AuthConfiguration to enable custom claim validation.
-            services.AddSingleton<AuthenticationConfiguration>();
+            // Register AuthConfiguration to enable custom claim validation for skills.
+            if (IsSkill(settings))
+            {
+                services.AddSingleton(sp => new AuthenticationConfiguration { ClaimsValidator = new AllowedCallersClaimsValidator(settings.Skill) });
```

Comment:
`settings` could be null, right?

Reply:
If yes, we cannot throw NullRef

Reply:
IsSkill returns false when null:
```csharp 
        public bool IsSkill(BotSettings settings)
        {
            return settings?.Skill?.IsSkill == true;
        }
```
So it should be fine